doc/install.rst

```diff
@@ -1,58 +1,655 @@
+    **cthread** Arbor          Default. Multithreaded, based on C++11 ``std::thread``.
+    **serial**  Arbor          Single threaded.
+    **tbb**     git submodule  `Intel TBB <https://www.threadingbuildingblocks.org/>`_.
+                               Recommended when using many threads.
+    =========== ============== =================================================
+
+.. Note::
+    The default `cthread` threading is suitable for most applications.
+    However there are some situations when the overheads of the threading runtime
+    become significant. This is often the case for:
+
+    * simulations with many small/light cells (e.g. LIF cells);
+    * running with many threads, such as on IBM Power 8 (80 threads/socket) or Intel
+      KNL (64-256 threads/socket).
+
+    The TBB threading back end is highly optimized, and well suited to these cases.
+
+
+.. Note::
+    If the TBB back end is selected, Arbor's CMake uses a git submodule of the TBB
+    repository to build and link a static version of the the TBB library. If you get
+    an error stating that the TBB submodule is not available, you must update the git
+    submodules:
+
+    .. code-block:: bash
+
+        git submodule update --init --recursive
+
+.. Note::
+    The TBB back end can be used on IBM Power 8 systems.
+
+.. _gpu:
+
+GPU Backend
+-----------
+
+Arbor supports NVIDIA GPUs using CUDA. The CUDA back end is enabled by setting the CMake ``ARB_WITH_CUDA`` option.
+
+.. code-block:: bash
+
+    cmake .. -DARB_WITH_CUDA=ON
+
+.. Note::
+    Abor requires CUDA version >= 8, and targets P100 GPUs.
+
+.. _cluster:
+
+HPC Clusters
+============
+
+HPC clusters offer their own unique challenges when compiling and running
+software, so we cover some common issues in this section.  If you have problems
+on your target system that are not covered here, please make an issue on the
+Arbor `Github issues <https://github.com/eth-cscs/arbor/issues>`_ page.
+We will do our best to help you directly, and update this guide to help other users.
+
+MPI
+---
+
+Arbor uses MPI for distributed systems. By default it is built without MPI support, which
+can enabled by setting the ``DARB_DISTRIBUTED_MODEL`` CMake parameter.
+An example of building Arbor with MPI, high-performance threading and optimizations enabled
+is:
+
+.. code-block:: bash
+
+    # set the compiler wrappers
+    export CC=`which mpicc`
+    export CXX=`which mpicxx`
+
+    # configure with mpi, tbb threading and compiled with optimizations
+    cmake .. -DARB_DISTRIBUTED_MODEL=mpi \      # Use MPI
+             -DCMAKE_BUILD_TYPE=release  \      # Optimizations on
+             -DARB_THREADING_MODEL=tbb   \      # TBB threading library
+
+    # run unit tests for global communication on 2 MPI ranks
+    mpirun -n 2 ./tests/global_communication.exe
+
+The first step to building with MPI support is to set the ``CC`` and ``CXX`` environment variables to refer to the mpi compiler wrappers.
+
```

Comment:
Out of the box, we don't need to use wrappers (e.g. bog standard Linux build). I think it would be useful to clarify that on some HPC systems one might need to use the wrappers in order to link against the right MPI libraries or whatever.

Reply:
95% of the time people use compiler wrappers, particularly if compiling on a cluster.
I have amended things to point out that this example uses compiler wrappers, but that it is perfectly possible to compile without them.

Reply:
Right -- the mpi support in cmake is horrible because it tries to deconstruct the black boxes of the wrappers into cmake data structures by probing them, and then reconstruct cmake command lines that do the same thing as the mpiwrappers would -- obviously impossible in general.

So maybe point out that it is highly suggested to **not** use cmake mpi deduction, but just use the mpi wrapper as the compiler (not said explicitly).